cmd/prestop/main.go

```diff
@@ -203,24 +187,72 @@ func isWaitingToBeDeleted(healthStatus agenthealth.MmsDirectorStatus) bool {
 	}
 	lastPlan := healthStatus.Plans[len(healthStatus.Plans)-1]
 	for _, m := range lastPlan.Moves {
-
-		// The next conditions are based on observations on the outcome
-		// of the agent after they have stopped the mongo server.
-
+		// When changing version the plan will contain a "ChangeVersion" step
 		switch m.Move {
-		case "WaitFeatureCompatibilityVersionCorrect":
-			// First condition observed. This is the Plan reported by the
-			// agent on the first MongoD stopped.
-			for _, s := range m.Steps {
-				if s.Step == "WaitFeatureCompatibilityVersionCorrect" &&
-					s.Result == "" {
-					return true
-				}
-			}
 		case "ChangeVersion":
-			// This is the condition observed in the 2nd and 3rd Pods.
 			return true
```

Comment:
with a switch statement with a single case, I feel like a simple 

`if m.Move == "ChangeVersion"` is more straightforward

Reply:
Of course, that's much better! Fixed now, thanks!